siti_tools/siti.py

```diff
@@ -482,19 +482,24 @@ def eotf_hlg(
         Returns:
             frame_data: pixel values in the physical luminance up to 10,000 cd/m^2
         """
-        frame_data = np.maximum(frame_data, 0.0)
-        frame_data = np.minimum(frame_data, 1.0)
 
         a = 0.17883277
         b = 0.02372241
         c = 1.00429347
-        gamma = 1.2
+        if l_max <= 1000.0:
+            gamma = 1.2
+        else:
+            gamma = 1.2 + 0.42 * np.log10(l_max/1000.0)
 
         frame_data = (frame_data <= 0.5) * (np.power(frame_data, 2.0) / 3.0) + (
             frame_data > 0.5
         ) * (np.exp((frame_data - c) / a) - b)
 
-        frame_data = (l_max - l_min) * np.power(frame_data, gamma - 1.0) + l_min
+        # NOTE: The following is only valid when the function is applied to the Y channel.
+        # If frame_data was a color channel (R, G, or B), the correct implementation would be:
+        # (l_max - l_min) * np.power(Y, gamma - 1.0) * frame_data + l_min,
+        # where Y would be obtained from linearized R, G, B channels.
+        frame_data = (l_max - l_min) * np.power(frame_data, gamma) + l_min
```

Comment:
If `frame_data = Y` the equation in the NOTE can be written as:

`(l_max - l_min) * np.power(frame_data, gamma - 1.0 + 1.0) + l_min`

Reply:
This part I don't see in BT.2100, but to be honest I have only looked very briefly.

Reply:
@slhck You are right, it is not in the recommendation as it only defines the functions for the individual components. The "most correct" way would therefore be to convert the YUV input to RGB, apply the EOTF on individual components, and convert back to YUV. But I believe that is not really needed for our use case and this solution seems like a reasonable approximation. Would you agree?

Reply:
I see, yes, that makes sense. I was more confused about the subtraction `gamma - 1.0` in the original code and where that came from.